pandas/core/common.py

```diff
@@ -2439,6 +2439,8 @@ def _get_dtype_type(arr_or_dtype):
         return np.dtype(arr_or_dtype).type
     elif isinstance(arr_or_dtype, CategoricalDtype):
         return CategoricalDtypeType
+    elif arr_or_dtype is NotImplemented:
+        return type(NotImplemented)
```

Comment:
Makes sense, I'll make this change in is_bool_dtype instead ... ccing @jreback for reference